framework/src/modules/chain/index.js

```diff
@@ -61,6 +61,7 @@ module.exports = class ChainModule extends BaseModule {
 			'delegates:fork',
 			'loader:sync',
 			'dapps:change',
+			'invalidate_cache',
```

Comment:
I don't think we need to create a new event. The chain module shouldn't worry about firing a new event for invalidating cache. In case of a block addition/deletion, it should simply call: 
```
library.channel.publish('chain:blocks:change', block);
```
And the http_api module should decide based on this event whether it needs to invalidate cache or not.

Reply:
I like this idea but aren't we potentially overloading the `chain:blocks:change` what if we want to take another action in the future for this message?

Reply:
I think that `chain:blocks:change` is an event. There can be multiple actions for one event. One action could be to reset the cache, another could be to update Lisk Nano etc.. If in the future we want to add another action, we can simply add another listener for this event.
Please note that I am not recommending to use the same action to perform all the tasks, I am suggesting to add multiple event handlers aka actions for `chain:blocks:change` event.

Reply:
Changed HTTP module to listen for `chain:blocks:change` for clearing cache `onNewBlock`